server.js

```diff
@@ -2568,7 +2568,146 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
-// Code Climate integration
+// Code Climate coverage (new)
+camp.route(/^\/codeclimate\/tc\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var userRepo = match[1];  // eg, `kabisaict/flow`.
+  var format = match[2];
+  request({
+      method: 'GET',
+      uri: 'http://api.codeclimate.com/v1/repos?github_slug=' + userRepo,
+      json: true
+  }, function (err, res, body) {
+    var badgeData = getBadgeData('coverage', data);
+
+    if (err != null) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    if (!body.data || body.data.length == 0 || !body.data.hasOwnProperty('attributes')) {
+      badgeData.text[1] = 'unknown';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    var options = {
+      method: 'HEAD',
+      uri: 'https://api.codeclimate.com/v1/badges/' + body.data[0].attributes.badge_token + '/test_coverage',
+    };
+
+    request(options, function(err, res) {
+      if (err != null) {
+        badgeData.text[1] = 'invalid';
+        sendBadge(format, badgeData);
+        return;
+      }
+
+      try {
+        var statusMatch = res.headers['content-disposition']
+                             .match(/filename=".*test_coverage-(.+)\.svg"/);
+        if (!statusMatch) {
+          badgeData.text[1] = 'unknown';
+          sendBadge(format, badgeData);
+          return;
+        }
+
+        var score = statusMatch[1].replace('-', '.');
+        badgeData.text[1] = score;
+
+        if (score == 'A') {
+          badgeData.colorscheme = 'brightgreen';
+        } else if (score == 'B') {
+          badgeData.colorscheme = 'green';
+        } else if (score == 'C') {
+          badgeData.colorscheme = 'yellow';
+        } else if (score == 'D') {
+          badgeData.colorscheme = 'orange';
+        } else {
+          badgeData.colorscheme = 'red';
+        }
+        sendBadge(format, badgeData);
+      } catch(e) {
```

Comment:
Is this the "project not found" case? Is it possible to test that more explicitly above, so you can reserve the `catch` for "something went wrong" and return **invalid** instead of **not found**?

Reply:
I'm not exactly sure... I copied these code blocks from the existing CodeClimate blocks. Could you elaborate on when "not found" vs. "invalid" vs. "unknown" should be displayed?

Reply:
- When someone is asking for a nonexistent project, we should display something really clear and helpful like **project not found**, which tells them they have spelled something wrong, the project isn't public, etc.
- On the other hand **unknown** would mean CodeClimate has the project but for some reason can't answer the question, like some kind of error occurred when trying to compute it.
- And **invalid** is like a 500 error: something went wrong on our side or their side when we tried to fetch the information.

Reply:
Ah, I gotcha. I tried to fixed the errors to be more appropriate, so take a look when you can.